### PR TITLE
[runtime] Use MonoError in mono_rand_try_get_bytes

### PR DESCRIPTION
**Pull request by lambdageek**

### CODE REVIEW

File: mono/utils/mono-rand.c [hunk 1]
```diff
@@ -277,6 +290,7 @@ mono_rand_try_get_bytes (gpointer *handle, guchar *buffer, gint buffer_size)
 					continue;
 				g_warning("Entropy error! Error in read (%s).", strerror (errno));
 				/* exception will be thrown in managed code */
+				mono_error_set_generic_error (error, "System", "ExecutionEngineException", "Entropy error! Error in read");
```
Comment (alexrp) [inline mono/utils/mono-rand.c]: this should carry the `strerror (errno)` result

**Comment by alexrp** [comment]
Actually, I guess my previous comment applies to most of these cases. We should really propagate the full error information with the exceptions.

**Comment by lambdageek** [comment]
Good point, thanks @alexrp !

**Comment by alexrp** [comment]
LGTM but @ludovic-henry might want to have a look too.

File: mono/metadata/threadpool-ms.c [hunk 2]
```diff
@@ -227,9 +227,11 @@ rand_create (void)
 static guint32
 rand_next (gpointer *handle, guint32 min, guint32 max)
 {
+	MonoError error;
 	guint32 val;
-	if (!mono_rand_try_get_uint32 (handle, &val, min, max)) {
+	if (!mono_rand_try_get_uint32 (handle, &val, min, max, &error)) {
 		// FIXME handle error
+		mono_error_cleanup (&error);
 		g_assert_not_reached ();
```
Comment (ludovic-henry) [inline mono/metadata/threadpool-ms.c]: You could replace that by `mono_error_assert_ok(&error)`, so we know what was the error. It's going to assert anyway.

**Comment by lambdageek** [comment]
@monojenkins merge

**Pull request merged**